### ISSUE TITLE
Consent V2

### ISSUE THREAD

**Issue by lexislav** [elleracompany/craft-cookie-consent#106]
Any plans to update this plugin for Consent V2 requirements?

**Comment by mazaalai** [comment]
Yes, My clients are also asking for this to be implemented in their sites. It is effective as of this month.

https://cookiefirst.com/google-consent-mode-v2-released/

**Comment by robbygeybels** [comment]
@jellingsen any update?

**Comment by Toni-hr98** [comment]
+1

**Comment by jawadharaba98** [comment]
+1

**Comment by jellingsen** [comment]
Hi,

I can see that this is a highly wanted feature, but we're currently swamped and can't promise a date.
Hopefully getting to take a look at this in a couple of weeks, and then estimate if and when this will be implemented.

If you require this functionality urgently I would recommend you to look at other plugins or solutions.

**Comment by wmdhosting** [comment]
any chance to just get it >> Craft 5 support  and then add featuers.

**Comment by robbygeybels** [comment]
hi @jellingsen, this plugin is still available for sale in the store, with a recurring charge on it.  But it is useless for now. This is a high priority to FIX it now. If you're selling a product you should release a fix of this in advance and not leave people who supported you and bought this plugin out in the cold.

**Comment by wmdhosting** [comment]
will be great just to get  Craft 5 version..  . **thats not time intesive**   

SO we have at least what was till now.. on site.... then you have time for other things.

**Comment by wmdhosting** [comment]
Perhaps this can help. for future. improvents.. 
https://playground.cookieconsent.orestbida.com/

**Comment by jellingsen** [comment]
I've been looking into Google Consent Mode V2, and while google tag manager might be the standard, this is a vendor specific solution and some custom implementation should be expected. We might implement this in the plugin, but we need to cover all different use-cases and settings - and for now I suggest you implement a custom solution covering your own needs.

Looking at [this video](https://www.youtube.com/watch?v=yZjGzfWDc0Y) the implementation in tag manager seems fairly trivial for marketers, and for developers this can also be implemented in the java script logic.

**Comment by wmdhosting** [comment]
OK. but will you make this plugin support craft 5 as that not big deal 
... then make changes..

**Comment by wmdhosting** [comment]
**just info if helps anything to speed up development.**.  for this same cookie script..  https://playground.cookieconsent.orestbida.com/
Info posted on discord.. 

croxton — 
I've been using a vanilla js cookie banner with some code to update the Google Consent Mode v2 state when the user changes their consent. This works great with Google Consent Mode v2 aware tags like Google Analytics, just like it does with an official CMP like Cookiebot. The only thing to be aware of is that you will need to use an official CMP if you want to actually serve ads on your site. Here's a example: https://gist.github.com/croxton/060899d07a2a9491ab0c20da74fdb700

**Comment by robbygeybels** [comment]
Really sad that this hasn't been picked up yet for a plugin we paid for...

**Comment by wmdhosting** [comment]
Any update .. on this ..

**Comment by wmdhosting** [comment]
I see there where some updates.. to plugin
Is this will go anytime soon..its 8 months now..
https://playground.cookieconsent.orestbida.com/

**Comment by MattAppleton** [comment]
Any news — we were considering this plugin — until we saw this! The ONLY cookies we actually need are Google Analytics...

**Comment by jellingsen** [comment]
Google Consent Mode v2 support is now implemented in version 5.2.0.

How it works:

1. **Map your cookie groups to consent signals** on each group's settings page (new installs get the typical mappings automatically: Statistics → `analytics_storage`, Marketing → `ad_storage`/`ad_user_data`/`ad_personalization`).
2. **Enable Consent Mode** in the plugin's site settings.
3. **Add one line at the top of your `<head>`**, before your gtag.js/GTM snippet:

```twig
{{ craft.cookieConsent.consentModeScript() }}
```

The script sets the default consent state (denied) synchronously before any Google tags load, then the visitor's stored consent is pushed as a `gtag('consent', 'update', ...)` — including immediately when they accept or change their consent in the banner, no reload needed.

A few notes:

- This is Google's **advanced** consent mode: your tags load normally and send cookieless pings until consent is granted, preserving conversion modeling. Basic mode (no tags before consent) remains possible via the `ccc-loaded` event — both are covered in the README.
- **SEOmatic users:** no extra configuration — SEOmatic injects its tracking scripts right before `</head>`, so the snippet above always runs first.
- Works with static page caches (Blitz): nothing visitor-specific is rendered server-side.
- Settings are available for the data layer name, `wait_for_update`, `ads_data_redaction` and `url_passthrough`.
